src/ui/common/src/components/workflows/workflowHeader.tsx

```diff
@@ -43,8 +51,12 @@ const WorkflowHeader: React.FC<Props> = ({ user, workflowDag }) => {
   const [showSuccessToast, setShowSuccessToast] = useState(false);
   const [showSettings, setShowSettings] = useState(false);
 
-  const handleSuccessToastClose = () => {
+  const handleSuccessToastClose = async () => {
     setShowSuccessToast(false);
+    await dispatch(handleGetWorkflow({ apiKey: user.apiKey, workflowId }));
```

Comment:
Will this block execution until the action get to `fulfilled` or `rejected` ?

Reply:
yes, it will wait till the handleGetWorkflow call completes before calling the next one, which is what we want to ensure we have our data ready.

One thing I could do is add a try catch here and handle errors a little better.